samples/pipeline/masstransit-messages/sample.md

```diff
@@ -0,0 +1,182 @@
+---
+title: Consuming messages from MassTransit
+summary: Use the NServiceBus pipeline to consume messages sent by MassTransit.
+reviewed: 2021-11-04
+component: Core
+related:
+ - nservicebus/pipeline
+ - nservicebus/pipeline/manipulate-with-behaviors
+---
+
+It's possible to migrate from a system built on MassTransit to NServiceBus without having to undergo the Big Rewrite. This sample will show how a messaging system can gradually be transitioned from MassTransit to NServiceBus a bit at a time by using an NServiceBus [message pipeline behavior](/nservicebus/pipeline/manipulate-with-behaviors.md) to translate incoming MassTransit messages into a shape that can be understood by NServiceBus.
+
+downloadbutton
+
+## Prerequisites
+
+This sample requires a local instance of RabbitMQ.
+
+## Message structure comparison
+
+When using a message structured like this:
+
+snippet: MassTransitEvent
+
+MassTransit will structure the message in RabbitMQ like this:
+
+**Properties**
+
+* **message_id**: 5fdc0000-426e-001c-fcf9-08d9a30339e8
+* **delivery_mode**: 2
+* **headers**:
+  * **Content**-Type: application/vnd.masstransit+json
+  * **publishId**: 9
+* **content_type**: application/vnd.masstransit+json
+
+**Payload**
+
+```json
+{
+  "messageId": "5fdc0000-426e-001c-fcf9-08d9a30339e8",
+  "conversationId": "5fdc0000-426e-001c-fda0-08d9a30339e8",
+  "sourceAddress": "rabbitmq://hostos/MACHINENAME_MTEndpoint_bus_m9qyyynnpayb3rk1bdc4gy3wyc?temporary=true",
+  "destinationAddress": "rabbitmq://hostos/Messages.Events:MassTransitEvent",
+  "messageType": [
+    "urn:message:Messages.Events:MassTransitEvent"
+  ],
+  "message": {
+    "text": "The time is 11/8/2021 4:00:50 PM -06:00"
+  },
+  "sentTime": "2021-11-08T22:00:50.1435641Z",
+  "headers": {},
+  "host": {
+    "machineName": "MACHINENAME",
+    "processName": "MTEndpoint",
+    "processId": 13892,
+    "assembly": "MTEndpoint",
+    "assemblyVersion": "1.0.0.0",
+    "frameworkVersion": "5.0.11",
+    "massTransitVersion": "7.2.4.0",
+    "operatingSystemVersion": "Microsoft Windows NT 10.0.19043.0"
+  }
+}
+```
+
+By contrast, NServiceBus structures a message on RabbitMQ quite a bit differently, preferring to reserve the payload entirely for the message body, and putting all of the metadata into RabbitMQ properties. If NServiceBus had published the event, it would look like this:
```

Comment:
```suggestion
By contrast, NServiceBus structures a message on RabbitMQ differently, reserving the payload entirely for the message body, and putting the metadata into RabbitMQ properties. If NServiceBus had published the event, it would look like this:
```